packages/blade/src/components/Tooltip/__tests__/Tooltip.web.test.tsx

```diff
@@ -0,0 +1,250 @@
+    });
+    expect(screen.queryByRole('tooltip')).toBeInTheDocument();
+
+    await act(async () => {
+      jest.advanceTimersByTime(1);
+    });
+
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+  });
+
+  it('should open/close on focus/blur with non-interactive trigger element', async () => {
+    const { getByTestId } = renderWithTheme(
+      <Tooltip content="Hello world">
+        <TooltipInteractiveWrapper>
+          <InfoIcon color="surface.action.icon.default.highContrast" size="medium" />
+        </TooltipInteractiveWrapper>
+      </Tooltip>,
+    );
+    await waitForPosition();
+
+    const wrapper = getByTestId('tooltip-interactive-wrapper');
+    fireEvent.focus(wrapper);
+
+    // on focus it should immediately open
+    await act(async () => {
+      jest.advanceTimersByTime(1);
+    });
+    expect(screen.queryByRole('tooltip')).toBeInTheDocument();
+
+    // close
+    fireEvent.blur(wrapper);
+    await act(async () => {
+      jest.advanceTimersByTime(animationDuration);
+    });
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+  });
+
+  it('should work with custom trigger element', async () => {
+    const testId = 'custom-trigger';
+    const CustomTrigger = React.forwardRef<
+      HTMLDivElement,
+      { children: React.ReactNode } & BladeCommonEvents
+    >(
+      (
+        {
+          children,
+          onBlur,
+          onFocus,
+          onMouseLeave,
+          onMouseMove,
+          onPointerDown,
+          onPointerEnter,
+          onTouchEnd,
+          onTouchStart,
+        },
+        ref,
+      ) => {
+        return (
+          <BaseBox
+            ref={ref}
+            tabIndex={0}
+            onBlur={onBlur}
+            onFocus={onFocus}
+            onMouseLeave={onMouseLeave}
+            onMouseMove={onMouseMove}
+            onPointerDown={onPointerDown}
+            onPointerEnter={onPointerEnter}
+            onTouchEnd={onTouchEnd}
+            onTouchStart={onTouchStart}
+            data-testid={testId}
+          >
+            <Text contrast="low">{children}</Text>
+          </BaseBox>
+        );
+      },
+    );
+
+    const { getByTestId } = renderWithTheme(
+      <Tooltip content="Hello world">
+        <CustomTrigger>Custom Trigger</CustomTrigger>
+      </Tooltip>,
+    );
+    await waitForPosition();
+
+    fireEvent.mouseEnter(getByTestId(testId));
+
+    // should not be opened by default
+    await act(async () => {
+      jest.advanceTimersByTime(1);
+    });
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+    // should open after 300ms
+    await act(async () => {
+      jest.advanceTimersByTime(299);
+    });
+    expect(screen.queryByRole('tooltip')).toBeInTheDocument();
+
+    // close
+    fireEvent.blur(getByTestId(testId));
+    await act(async () => {
+      jest.advanceTimersByTime(animationDuration);
+    });
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+  });
+
```

Comment:
can we add a test to check for meta attributes so we are assured that our analytics won't be missed?

Reply:
Done. But we need a reusable solution to this similar to "assertAccessible"